### PR TITLE
690 Installing the Docker moby-buildx on Ubuntu Images

### PR DESCRIPTION
**Pull request by andy-mishechkin**
# Description
New tool.
According to https://github.com/actions/virtual-environments/issues/690, moby-buildx tool has been added to docker-moby.sh
Successful builds:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=65567&view=results
https://github.visualstudio.com/virtual-environments/_build/results?buildId=65566&view=results

## Check list
- [X] Related issue / work item is attached
- [ X] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Comment by al-cheb** [comment]
Where do we have simple test to validate moby-buildx plugin?

**Comment by al-cheb** [comment]
Where do we update the documentation file to include moby-buildx version?

File: images/linux/scripts/installers/docker-moby.sh [hunk 1]
```diff
@@ -16,6 +16,7 @@ if ! IsInstalled $docker_package; then
     apt-get remove -y moby-engine moby-cli
     apt-get update
     apt-get install -y moby-engine moby-cli
+    apt-get install --no-install-recommends -y moby-buildx
```
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:19]: 1. Where do we have a simple test to validate moby-buildx plugin installation?
2. Where do we update the documentation file to include moby-buildx version?

File: images/linux/scripts/installers/docker-moby.sh [hunk 2]
```diff
@@ -50,3 +54,7 @@ docker pull ubuntu:14.04
 echo "Documenting Docker version"
 docker_version=$(docker -v)
 DocumentInstalledItem "Docker-Moby ($docker_version)"
+
+echo "Documenting moby-buildx version"
+moby_buildx_version=$(docker buildx -v)
+DocumentInstalledItem "Moby-Buildx ($moby_buildx_version)"
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/docker-moby.sh:60]: May be `Docker-Buildx` to be consistent with `Docker-Moby`?

File: images/linux/scripts/installers/docker-moby.sh [hunk 3]
```diff
@@ -16,16 +16,22 @@ if ! IsInstalled $docker_package; then
...
     echo "Docker ($docker_package) is already installed"
 fi
 
 # Run tests to determine that the software installed as expected
 echo "Testing to make sure that script performed as expected, and basic scenarios work"
+echo "checking the docker-moby and moby-buildx"
 if ! command -v docker; then
     echo "docker was not installed"
     exit 1
+elif ! command -v docker buildx build; then
```
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:30]: This comparison contains logical error: we can have docker and missing buildx validation

File: images/linux/scripts/installers/docker-moby.sh [hunk 4]
```diff
@@ -50,3 +56,7 @@ docker pull ubuntu:14.04
 echo "Documenting Docker version"
 docker_version=$(docker -v)
 DocumentInstalledItem "Docker-Moby ($docker_version)"
+
+echo "Documenting Docker-buildx version"
+docker_buildx_version=$(docker buildx -v)
```
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:61]: docker buildx doesn't have flag -v

File: images/linux/scripts/installers/docker-moby.sh (hunk 3 above)
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:30]: Syntax is incorrect to check subcommand:
```
$ command -v docker NotFound
/usr/bin/docker
```

File: images/linux/scripts/installers/docker-moby.sh [hunk 5]
```diff
@@ -50,3 +57,8 @@ docker pull ubuntu:14.04
 echo "Documenting Docker version"
 docker_version=$(docker -v)
 DocumentInstalledItem "Docker-Moby ($docker_version)"
+
+echo "Documenting Docker-buildx version"
+DOCKER_BUILDX_VERSION=$(apt-cache policy moby-buildx | grep Installed)
```
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:62]: Why do we use so complex syntax?
`docker buildx version`

File: images/linux/scripts/installers/docker-moby.sh [hunk 6]
```diff
@@ -16,16 +16,23 @@ if ! IsInstalled $docker_package; then
...
 fi
 
 # Run tests to determine that the software installed as expected
 echo "Testing to make sure that script performed as expected, and basic scenarios work"
+echo "Checking the docker-moby and moby-buildx"
+DOCKER_BUILDX=/usr/libexec/docker/cli-plugins/docker-buildx
 if ! command -v docker; then
     echo "docker was not installed"
     exit 1
+elif ! [ -x "$DOCKER_BUILDX" ]; then
```
Comment (al-cheb) [inline images/linux/scripts/installers/docker-moby.sh:31]: Why do we need to hardcoded the path if we can test the `docker buildx` command result?

**Comment by andy-mishechkin** [comment]
`testAdm2@danurgDocker:~$ command -v docker
/usr/bin/docker
testAdm2@danurgDocker:~$ command -v docker buidx
/usr/bin/docker
testAdm2@danurgDocker:~$ command -v docker buidxxx
/usr/bin/docker`

command -v doesn't work for docker plugins

**Comment by andy-mishechkin** [comment]
`testAdm2@danurgDocker:~$ docker buildx version

github.com/docker/buildx 0.3.1+azure 6db68d029599c6710a32aa7adcba8e5a344795a7`

File: images/linux/scripts/installers/docker-moby.sh (hunk 5 above)
Comment (andy-mishechkin) [reply images/linux/scripts/installers/docker-moby.sh:62]:
```
testAdm2@danurgDocker:~$ command -v docker
/usr/bin/docker

testAdm2@danurgDocker:~$ command -v docker buidx
/usr/bin/docker

testAdm2@danurgDocker:~$ command -v docker buidxxx
/usr/bin/docker
```

command -v doesn't work for docker plugins

Comment (andy-mishechkin) [reply images/linux/scripts/installers/docker-moby.sh:62]:
```
testAdm2@danurgDocker:~$ docker buildx version

github.com/docker/buildx 0.3.1+azure 6db68d029599c6710a32aa7adcba8e5a344795a7
```
this is more hard for parsing then `apt-cache policy moby-buildx | grep Installed` result

File: images/linux/scripts/installers/docker-moby.sh (hunk 6 above)
Comment (andy-mishechkin) [reply images/linux/scripts/installers/docker-moby.sh:31]:
```
testAdm2@danurgDocker:~$ command -v docker
/usr/bin/docker

testAdm2@danurgDocker:~$ command -v docker buidx
/usr/bin/docker

testAdm2@danurgDocker:~$ command -v docker buidxxx
/usr/bin/docker
```

command -v doesn't work for docker plugins

Comment (al-cheb) [reply images/linux/scripts/installers/docker-moby.sh:31]: Why do we need to use the `command` to validate docker buidx? Check the exit code.

File: images/linux/scripts/installers/docker-moby.sh (hunk 5 above)
Comment (al-cheb) [reply images/linux/scripts/installers/docker-moby.sh:62]: Really `docker buildx version | cut -d ' ' -f2` ?

Comment (andy-mishechkin) [reply images/linux/scripts/installers/docker-moby.sh:62]: why didn't you immediately provide your best approach without long conversation ?

File: images/linux/scripts/installers/docker-moby.sh (hunk 6 above)
Comment (andy-mishechkin) [reply images/linux/scripts/installers/docker-moby.sh:31]: If you have a best approach - just provide it here without many words

Comment (al-cheb) [reply images/linux/scripts/installers/docker-moby.sh:31]: ! [[ $(docker buildx) ]]

**Pull request merged**